app/services/metrics/base_metric_processor.rb

```diff
@@ -0,0 +1,63 @@
+module Metrics
```

Comment:
I think we could add some tests for this file too, so we don't depend on tests for the specific implementations